lib/stat_tracker.rb

```diff
@@ -23,6 +23,41 @@ def initialize(files)
     end
   end
 
+  def all_season_game_id(season)
+    @games.map do |game|
+      game.game_id if game.season == season
+    end.compact 
+  end 
+
+  def winningest_coach(season)
+    # we want the winning coach with the best win percentage in a season
+    # output: string
+    # 1. we will used games.csv and match the season with our input season, and return an array of all games in that season
+    # 2. iterate over game_teams.csv using our created array in all_season_game_id as referance
+    #       make a hash of our head coach as key and set values to [wins, total_games]
+    # 3 transform_values on our created hash above and reset the value to wins/total games.round(4)
+    # 4 after we have our array of keys being couch names and use .values to make value array, then .max array, then .keys(max_value) to get coach name
+
+    coach_wins = @game_teams.each_with_object(Hash.new([0,0])) do |game, hash|
+
+      if all_season_game_id(season).include?(game.game_id) 
+        if game.result == "WIN"
+          hash[game.head_coach] = [1 + hash[game.head_coach][0], 1 + hash[game.head_coach][1]]
```

Comment:
I like the similar use of using a two element array as values similar to Charles, method developments.